src/core/Track.cpp

```diff
@@ -1896,6 +1895,22 @@ bool TrackContentWidget::pasteSelection( MidiTime tcoPos, const QMimeData * md,
 	// The offset is quantized (rather than the positions) to preserve fine adjustments
 	offset = offset.quantize(snapSize);
 
+	// Get the leftmost TCO and fix the offset if it reaches below bar 0
+	MidiTime leftmostPos = grabbedTCOPos;
+	for(int i = 0; i < tcoNodes.length(); ++i)
+	{
+		QDomElement outerTCOElement = tcoNodes.item(i).toElement();
+		QDomElement tcoElement = outerTCOElement.firstChildElement();
+
+		MidiTime pos = tcoElement.attributeNode("pos").value().toInt();
+
+		if(pos < leftmostPos) { leftmostPos = pos; }
+	}
+	if((leftmostPos + offset) < 0)
+	{
+		offset = -leftmostPos;
+	}
+
```

Comment:
```suggestion
	offset = max(offset, -leftmostPos)
```

Reply:
True, that works! It makes the logic a tiny bit less explicit, but works just the same. Should I use `max` or `qMax`?

Reply:
No preference!

Reply:
Used `std::max`. `qMax` is more often used, but from now on I think I'll give preference to `std` unless it's a requirement that we have some Qt feature. I didn't use max on the MoveSelection action because IMHO it needs a refactor for better readability, so this could be changed when that happen.

I just wanted to explain why I explicitly called `getTicks` on both terms of `std::max`:
If I used `std::max(offset, -leftmostPos);` I'd be comparing a `MidiTime` with an `int` type (compiler error), because since there's an arithmetic operation on `leftmostPos` it calls the `operator int()` method returning `m_ticks` which happens to be an `int` type. If I have `std::max(offset.getTicks(), -leftmostPos);` it would work fine, but I feel like it's bad practice to compare `ticks_t` with `int`, even though they are matching types, without explicitly showing what types are being compared. If it sounds like an overkill I can replace `-leftmostPos.getTicks()` with just `-leftmostPos`.

Reply:
I think the only issue with `std::max` is Windows will throw errors sometimes, at least that's what always seems to happen to me. lol

Reply:
Since CI is still passing, I guess it's fine in this case.